js/src/common/components/Page.js

```diff
@@ -12,15 +12,18 @@ export default class Page extends Component {
 
     this.onNewRoute();
 
-    app.drawer.hide();
-    app.modal.close();
-
     /**
      * A class name to apply to the body while the route is active.
      *
      * @type {String}
      */
     this.bodyClass = '';
+
+    this.currentPath = m.route.get();
+  }
+
+  routeHasChanged() {
```

Comment:
Do we want this to be public API, or could we just do the test in `onbeforeupdate` ?

Reply:
I think public API is good: subclasses with more complex use cases (like DiscussionPage) could use it to do more of those checks, and also possibly handle some side effects if the route hasn't changed but we still want to do something (like scrolling to a post).